pkg/cli/options.go

```diff
@@ -155,8 +155,18 @@ func WithCompletion() Option {
 
 // parseExternalPluginArgs returns the program arguments.
 func parseExternalPluginArgs() (args []string) {
-	args = make([]string, len(os.Args)-1)
```

Comment:
I would like to see this scenarios tested with e2e. Example:

By having the sample plugin here and calling them as we do in the others e2e tests.
So that we can ensure the expected results, which would make it simple, we review the changes.

could we try to work on that ? WDYT @rashmigottipati @everettraven ?

Reply:
@camilamacedo86 I'll work on a separate PR to add e2e test with the python external plugin I've built. But I wouldn't make e2e tests as a part of this PR, as the goal of this PR is to enable CLI flags for external plugins.

Reply:
I'd be happy to go through and add as a comment to this thread the reasoning for this change with proof when running it locally.

Reply:
When using the original method (adding a print statement to print the args) with the changes that pass the CLI flags to an external plugin, running `kubebuilder init --plugins myexternalplugin/v1 --domain example.com` results in this error output:
```
[init --plugins myexternalplugin/v1 --domain example.com]
INFO[0000] Adding external plugin: myexternalplugin     
usage: myexternalplugin.py [-h] [--domain DOMAIN] [--license LICENSE]
                           [--boolean | --no-boolean]
myexternalplugin.py: error: unrecognized arguments: init --plugins myexternalplugin/v1
Error: failed to initialize project: unable to scaffold with "myexternalplugin/v1": exit status 2
Usage:
  kubebuilder init [flags]

Examples:
  # Help for initializing a project with version "2"
  kubebuilder init --project-version="2" -h

  # Help for initializing a project with version "3"
  kubebuilder init --project-version="3" -h

Flags:
      --boolean                  flag to test boolean logic
      --domain string            Project domain (default "my.domain")
      --float float              flag to test float logic (default 123.45)
  -h, --help                     help for init
      --integer int              flag to test integer logic (default 123)
      --license string           Project license (default "apache2")
      --project-version string   project version (default "3")

Global Flags:
      --plugins strings   plugin keys to be used for this subcommand execution

2022/06/01 09:18:05 failed to initialize project: unable to scaffold with "myexternalplugin/v1": exit status 2

```

The important line to note in that error message is:
> myexternalplugin.py: error: unrecognized arguments: init --plugins myexternalplugin/v1

This tells us that we are parsing all arguments after the program name itself, including subcommands and flags that are likely only to be used with Kubebuilder (`--plugins` flag specifically). We can see this more so from the print statement that I added for testing here:
> [init --plugins myexternalplugin/v1 --domain example.com]

The modified function ensures that we only parse flags and the corresponding flag values. It also ensures that we do not parse the `--plugins` flag since it is likely only used with Kubebuilder.

With the new changes the output of the same command is:
```
[--domain example.com]
INFO[0000] Adding external plugin: myexternalplugin  
```

There is no error since the proper flags are passed to the external plugin.

@camilamacedo86 @rashmigottipati I hope this helps demonstrate the reason for the change!